server/container/guice/guice-common/src/main/java/org/apache/james/utils/InMemoryMailRepositoryStore.java

```diff
@@ -103,11 +104,10 @@ public void init() throws Exception {
 
     @Override
     public Stream<MailRepository> getByPath(MailRepositoryPath path) {
```

Comment:
Reading the javadoc, everything is wrong : 
```
    /**
     * Returns the {@link MailRepository} for the given url.
     * This mail repository will not be created if it does not exist.
     */
    Optional<MailRepository> get(MailRepositoryUrl url) throws MailRepositoryStoreException;

    /**
     * Returns all the {@link MailRepository} for the given path.
     */
    Stream<MailRepository> getByPath(MailRepositoryPath path) throws MailRepositoryStoreException;
```

get by Url is not supposed to call select because select create the repository if it doesn't exist, and the javadoc says the opposite.
getByPath should use the same pattern as get (byUrl) and maybe we need another set of methods (select*) for the behavior you want

Reply:
Arf. Here it get's tricky.

```
Scenario:

 - James server 1 creates "cassandra://var/mail/dlp/truc"
 - The URL is listed then when admin list repos on James server 2 he sees "var/mail/dlp/truc" path
 - He expects to interact (like view content) of  "var/mail/dlp/truc" path on James server 2
 - However "cassandra://var/mail/dlp/truc" repo do not exist on James server 2
 - We need on such occasion to CREATE  "cassandra://var/mail/dlp/truc" repo on James server 2 to allow that interaction.
```

Just like we do with get.

Please not that a **MailRepository** may not exist locally, but it exist as soon as its URL is listed in the URL store. Hence the behavior I implemented do not offend that Java doc.

Maybe the way to solve this is to add this disclaimer to the JavaDoc.